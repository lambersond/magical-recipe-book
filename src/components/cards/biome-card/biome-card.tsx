import { IngredientChip } from '@/components/chips'
import { Card } from '@/components/common'
import type { Biome } from '@/types'

export function BiomeCard({
  description,
  image,
  name,
  ingredients,
}: Readonly<Biome>) {
  return (
    <Card>
      <span className='flex gap-3 items-center mb-4'>
        <p className='text-4xl'>{image}</p>
        <span className='flex flex-col'>
          <p className='text-2xl'>{name}</p>
          <span className='flex gap-1 text-text-secondary'>
            <p className='font-bold'>Ingredients:</p>
            <p>{ingredients?.length}</p>
          </span>
        </span>
      </span>
      <p className='text-text-secondary mb-4 border-b border-border pb-4'>
        {description}
      </p>
      <div className='flex gap-1 flex-wrap'>
        {ingredients?.map(({ ingredient }) => (
          <IngredientChip key={ingredient.id} {...ingredient} />
        ))}
      </div>
    </Card>
  )
}
