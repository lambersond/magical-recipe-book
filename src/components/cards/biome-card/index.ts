export { BiomeCard } from './biome-card'
