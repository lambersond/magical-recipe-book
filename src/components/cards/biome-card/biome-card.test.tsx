import { render } from '@test-utils'
import { BiomeCard } from './biome-card'

describe('components/cards/biome-card', () => {
  it('should match snapshot', () => {
    expect(
      render(
        <BiomeCard
          key='unused'
          id='abc'
          image='ðŸŒ‹'
          name='biomeName'
          description=''
          ingredients={[
            { ingredient: { id: '1', name: 'ingredient1', rarity: 'common' } },
          ]}
        />,
      ).asFragment(),
    ).toMatchSnapshot()
  })
})
