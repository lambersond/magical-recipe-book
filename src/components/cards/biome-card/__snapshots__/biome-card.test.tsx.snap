// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`components/cards/biome-card should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="p-4 border-border bg-card w-full rounded-none sm:rounded-lg"
    data-testid="Card"
  >
    <span
      class="flex gap-3 items-center mb-4"
    >
      <p
        class="text-4xl"
      >
        ðŸŒ‹
      </p>
      <span
        class="flex flex-col"
      >
        <p
          class="text-2xl"
        >
          biomeName
        </p>
        <span
          class="flex gap-1 text-text-secondary"
        >
          <p
            class="font-bold"
          >
            Ingredients:
          </p>
          <p>
            1
          </p>
        </span>
      </span>
    </span>
    <p
      class="text-text-secondary mb-4 border-b border-border pb-4"
    />
    <div
      class="flex gap-1 flex-wrap"
    >
      <div
        class="border-rarity-common bg-rarity-common/20 text-text-rarity-common text-xs rounded-sm px-2 py-1 border flex min-w-fit items-center"
        data-state="closed"
      >
        <svg
          class="hidden sm:inline sm:mr-2 size-4"
          data-testid="D4Icon"
          height="24"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 5120 5120"
          width="24"
        >
          <path
            d="M2349 5024 c-25 -38 -113 -157 -196 -264 -154 -202 -518 -680 -1064 -1400 -174 -228 -450 -593 -615 -810 -389 -512 -474 -626 -474 -639 0 -22 466 -396 1216 -976 50 -38 246 -190 435 -337 189 -146 434 -336 544 -421 l200 -154 6 496 c8 647 8 4026 0 4335 l-6 239 -46 -69z M2710 5080 c-5 -40 -16 -4669 -12 -4833 l7 -227 265 203 c146 112 421 325 611 473 587 457 1227 955 1387 1078 83 65 152 120 152 123 0 9 -566 759 -1465 1943 -157 206 -428 564 -603 794 -367 486 -341 452 -342 446z"
            fill="currentColor"
          />
        </svg>
        <p
          class=""
        >
          ingredient1
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
